llvm/lib/CodeGen/TwoAddressInstructionPass.cpp

```diff
@@ -203,26 +226,75 @@ class TwoAddressInstructionPass : public MachineFunctionPass {
     AU.addPreservedID(MachineDominatorsID);
     MachineFunctionPass::getAnalysisUsage(AU);
   }
-
-  /// Pass entry point.
-  bool runOnMachineFunction(MachineFunction&) override;
 };
 
 } // end anonymous namespace
 
-char TwoAddressInstructionPass::ID = 0;
+PreservedAnalyses
+TwoAddressInstructionPass::run(MachineFunction &MF,
+                               MachineFunctionAnalysisManager &MFAM) {
+  // Disable optimizations if requested. We cannot skip the whole pass as some
+  // fixups are necessary for correctness.
+  TwoAddressInstructionImpl Impl(MF, MFAM);
+  if (MF.getFunction().hasOptNone())
```

Comment:
shouldn't this be skipFunction the same way?

Reply:
skipFunction is legacy pass manager only and there is no counterpart in new pass manager, so here just check the optnone, like what skipFunction do. The rest part in skipFunction is in instrumentation.